app/stores/data-source.ts

```diff
@@ -0,0 +1,62 @@
+import create, { StateCreator, StoreApi } from "zustand";
+
+import { DataSource } from "@/configurator";
+import {
+  DEFAULT_DATA_SOURCE,
+  getURLParam,
+  parseDataSource,
+  stringifyDataSource,
+} from "@/domain/datasource";
+import { isRunningInBrowser } from "@/lib/is-running-in-browser";
+
+type DataSourceStore = {
+  dataSource: DataSource;
+  setDataSource: (value: string) => void;
```

Comment:
It's a bit unfortunate that we get passed a full DataSource but when we set, we set a string. It'd be good to have symmetry here, we receive a DataSource and set a DataSource.